ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor/State.hs

```diff
@@ -37,17 +41,57 @@ import Ouroboros.Network.Context
 import Ouroboros.Network.Mux
 
 
+-- | Public inbound governor state.
+--
+newtype PublicInboundGovernorState peerAddr versionData = PublicInboundGovernorState {
+      -- | A map of mature inbound duplex connections. These peers are used for
+      -- light peer sharing (e.g. bootstrapping peer sharing).
+      --
+      inboundDuplexPeers :: Map peerAddr versionData
+    }
+
+
+-- | Smart constructor for `PublicInboundGovernorState`.
+--
+-- NOTE: we assume that all inbound connections share the same address.  This
+-- is true in P2P mode since all outbound connections (which also can end up in
+-- the `InboundGovernorState`) bind the server address.  This allows us to use
+-- `Map.mapKeysMonotonic`.
+--
```

Comment:
I don't know what address you're referring to in this case but if this is the local IP address it isn't true. And it clearly isn't true for the remote address.

By default the node will bind to "any" address and a specific port number.